dotnet/Miscreant/IMac.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace Miscreant
+{
+	/// <summary>
+	/// Defines the basic operations of message authentication code.
+	/// </summary>
+	internal interface IMac : IDisposable
```

Comment:
It looks like there's a `KeyedHashAlgorithm` interface you can implement:

https://msdn.microsoft.com/en-us/library/system.security.cryptography.keyedhashalgorithm(v=vs.110).aspx

Reply:
I actually implemented the `KeyedHashAlgorithm` initially, but after performance testing I decided that it was not the best idea (see the commit 9b768d514f475ab2128f1238491347bfe40e96b7). The problem with `KeyedHashAlgorithm` is that it's not actually an interface, but an abstract class that does a whole lot unnecessary things behind the scenes. Also, given that the crypto part of the .NET standard library is stuck in the past, I though that we should't try too hard to be compatible with it.